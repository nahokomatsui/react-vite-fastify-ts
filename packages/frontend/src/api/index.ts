import aspida from "@aspida/axios";
import api from "./aspida/$api";
import axios from "axios";

const axiosInstance = () => {
  const baseURL = import.meta.env.VITE_API_BASE_URL;
  if (!baseURL) {
    throw new Error("env: VITE_API_BASE_URL must be set.");
  }
  return axios.create({
    baseURL,
  });
};

export const apiClient = api(aspida(axiosInstance()));
