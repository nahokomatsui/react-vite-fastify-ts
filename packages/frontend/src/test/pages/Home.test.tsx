import { describe, test, expect } from "vitest";
import { render } from "@testing-library/react";
import Home from "../../pages/Home";
import { pingResponse } from "../mocks/api/handlers";

describe("Home", () => {
  test("shows data from api successfully", async () => {
    const { asFragment, findByText } = render(<Home />);
    await findByText(pingResponse);
    expect(asFragment()).toMatchSnapshot();
  });
});
