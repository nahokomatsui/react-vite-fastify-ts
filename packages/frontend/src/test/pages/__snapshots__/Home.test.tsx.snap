// Vitest Snapshot v1

exports[`Home > shows data from api successfully 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <header
      class="Home-header"
    >
      <img
        alt="logo"
        class="Home-logo"
        src="/src/logo.svg"
      />
      <p>
        Hello Vite + React!
      </p>
      <p>
        Data from API: 
        <span
          style="font-weight: bold; color: greenyellow;"
        >
          pong!
        </span>
      </p>
      <p>
        <button
          type="button"
        >
          count is: 0
        </button>
      </p>
      <p>
        Edit 
        <code>
          App.tsx
        </code>
         and save to test HMR updates.
      </p>
      <p>
        <a
          class="Home-link"
          href="https://reactjs.org"
          rel="noopener noreferrer"
          target="_blank"
        >
          Learn React
        </a>
         | 
        <a
          class="Home-link"
          href="https://vitejs.dev/guide/features.html"
          rel="noopener noreferrer"
          target="_blank"
        >
          Vite Docs
        </a>
      </p>
    </header>
  </div>
</DocumentFragment>
`;
